compiler/rustc_traits/src/normalize_projection_ty.rs

```diff
@@ -40,7 +40,8 @@ fn normalize_canonicalized_projection_ty<'tcx>(
                 cause,
                 0,
                 &mut obligations,
-            );
+            )
+            .map_err(|_| NoSolution)?;
```

Comment:
The overflow would have errored out below anyway, causing `NoSolution`, so we're just doing this early now